pkg/subscriptions/subscriptions.js

```diff
@@ -0,0 +1,502 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* global jQuery   */
+
+define([
+    "jquery",
+    "base1/cockpit",
+    "base1/po",
+    "base1/mustache"
+], function($, cockpit, po, Mustache) {
+    "use strict";
+
+    cockpit.locale(po);
+    cockpit.translate();
+
+    function debug() {
+        if (window.debugging == "all" || window.debugging == "subscriptions")
+            console.debug.apply(console, arguments);
+    }
+
+    var subscriptions = { };
+
+    function register_system() {
+        var url = null;
+        if ($('#subscription-register-url').val() === 'Custom URL')
+          url = $('#subscription-register-url-custom').val().trim();
+        var username = $('#subscription-register-username').val().trim();
+        var password = $('#subscription-register-password').val().trim();
```

Comment:
> Not sure we want to trim passwords? Are you sure that passwords ending in a space are illegal?

You're absolutely right. It always annoys me when ui's make that kind of choice for me.